regex-assembly/942340.ra

```diff
@@ -1,28 +1,195 @@
 ##! Please refer to the documentation at
 ##! https://coreruleset.org/docs/development/regex_assembly/.
 
+##! The meaning of this rule is.. uncertain.
+##! Description doesn't help: "Detects basic SQL authentication bypass attempts 3/3"
+
 ##!+ i
 
+##! Definitions and reusables
+##!> define space_or_word_or_plus [\s\w+]+
+##!> define quotes [\"'`]
+##!> define digits \d+
+##!> define space \s+
+##!> define optional_space \s*?
+
+##! ^ is the same as xor in major databases
+##!> assemble
+  and
+  nand
+  or
+  xor
+  xxor
+  div
+  like
+  between
+  not
+  ##!=< logical_operators
+##!<
+
+##!> assemble
+  %
+  &
+  <
+  >
+  \^
+  =
+  ##!=< sql_operators
+##!<
+##! End definitions and reusables
+
+##!> assemble
+  \|\|
+  &&
+  ##!=< alternative_logical_operators
+##!<
+
 in\s*?\(+\s*?select
-(?i:(?:n?and|x?x?or|div|like|between|not)\s+|(?:\|\||\&\&)\s*)[\s\w+]+regexp\s*?\(
-(?i:(?:n?and|x?x?or|div|like|between|not)\s+|(?:\|\||\&\&)\s*)[\s\w+]+sounds\s+like\s*?[\"'`]
-(?i:(?:n?and|x?x?or|div|like|between|not)\s+|(?:\|\||\&\&)\s*)[\s\w+]+[=\d]+x
-[\"'`]\s*?\d\s*?--
-[\"'`]\s*?\d\s*?#
-[\"'`][\%&<>^=]+\d\s*?=
-[\"'`][\%&<>^=]+\d\s*?or
-[\"'`][\%&<>^=]+\d\s*?xor
-[\"'`][\%&<>^=]+\d\s*?div
-[\"'`][\%&<>^=]+\d\s*?like
-[\"'`][\%&<>^=]+\d\s*?between
-[\"'`][\%&<>^=]+\d\s*?and
-[\"'`]\W+[\w+-]+\s*?=\s*?\d\W+[\"'`]
-[\"'`]\s*?is\s*?\d.+[\"'`]?\w
-[\"'`]\|?[\w-]{3,}[^\w\s.,]+[\"'`]
-[\"'`]\s*?is\s*?[\d.]+\s*?\W.*?[\"'`]
-\bexcept\s+select\b
-\bexcept\s+values\s*?\(
-[\"'`]\s*(?i:(?:n?and|x?x?or|div|like|between|not)\s+|(?:\|\||\&\&)\s*)array\s*\[
-[\"'`]\s*(?i:(?:n?and|x?x?or|div|like|between|not)\s+|(?:\|\||\&\&)\s*)[\w]+\s*!?~
-[\"'`]\s*(?i:(?:n?and|x?x?or|div|like|between|not)\s+|(?:\|\||\&\&)\s*)[\w]+\s+(?:not\s+)?similar\s+to\s+
-[\"'`]\s*(?i:(?:n?and|x?x?or|div|like|between|not)\s+|(?:\|\||\&\&)\s*)(?:true|false)\b
+
+##!> assemble
+  ##!> assemble
+    ##!=> logical_operators
+    {{space}}
+  ##!<
+
+  ##!> assemble
+    ##!=> alternative_logical_operators
+    {{optional_space}}
+  ##!<
+  ##!=>
+  {{space_or_word_or_plus}}
+  ##!=>
+  regexp{{optional_space}}\(
+  sounds{{space}}like{{optional_space}}{{quotes}}
+  [=\d]+x
+##!<
+
+##! ORIGINAL :up:
```

Comment:
What is the meaning/context of these repeated `ORIGINAL :up:` comments?

Reply:
I think it should he `:down:` instead of `:up`. It is the original expression in the previous regex-assembly, for comparision.

Reply:
`:up:` because editors may interpret it as ☝️. "This is the original for the expression generated above".